src/cmd/faktory.go

```diff
@@ -185,15 +195,29 @@ func emitJobCompleteMetrics(jobStart time.Time, job opslevel.RunnerJob, outcome
 func legacyJobHandler(ctx context.Context, args ...interface{}) error {
 	jobStart := emitJobStartedMetrics()
 
-	job, err := parseJob(args)
-	if err != nil {
+	helper := worker.HelperFor(ctx)
+
+	var job opslevel.RunnerJob
+
+	if err := parseJob(args, &job); err != nil {
 		return err
 	}
 
-	helper := worker.HelperFor(ctx)
+	extractJobId(helper, &job)
 
-	job, err = prepareJob(helper, job)
-	if err != nil {
+	if err := extractCustomImage(helper, &job); err != nil {
+		return err
+	}
+
+	if err := extractCustomExtraCommands(helper, &job); err != nil {
+		return err
+	}
+
+	if err := extractCustomExtraVars(helper, &job); err != nil {
+		return err
+	}
+
+	if err := extractCustomExtraFiles(helper, &job); err != nil {
```

Comment:
This is really the meat of the change - i broke all the different things out into their own functions to effectively prepare the job by extracting the necessary data if its available. If any of these steps fail return an error so that we don't actually "run" the job.

This means in the future if we ever have to write different handlers we have the piecemeal parts to build up the job in that handler function.